criu/files-reg.c

```diff
@@ -2035,6 +2199,59 @@ static int validate_with_build_id(const int fd, const struct stat *fd_status,
 	return 1;
 }
 
+/*
+ * Compares the file's CRC32C checksum with the stored value.
+ * Returns 1 if the checksum of the file matches the checksum that was stored
+ * while dumping, -1 if there is a mismatch or 0 if the checksum has not been
+ * stored or could not be obtained.
+ */
+static int validate_with_checksum(const int fd, const struct stat *fd_status,
+					const struct reg_file_info *rfi)
+{
+	u32 checksum;
+
+	if (!rfi->rfe->has_size)
+		return 1;
+	
+	if (!rfi->rfe->has_checksum || !rfi->rfe->has_checksum_config ||
+		!rfi->rfe->has_checksum_parameter)
+		return 0;
+
+	/*
+	 * If the checksum configuration used while dumping is not the same as the current
+	 * checksum configuration, then the user is informed of this and the checksum
+	 * configuration used while dumping is considered.
+	 */
+	if (opts.file_validation_chksm_config != rfi->rfe->checksum_config) {
+		pr_warn("Checksum configuration (For validation) doesn't match the configuration used for dumping file %s\n",
+				rfi->path);
+		opts.file_validation_chksm_config = rfi->rfe->checksum_config;
+	}
+
+	/*
+	 * If the checksum parameter used while dumping is not the same as the current
+	 * checksum parameter, then the user is informed of this and the checksum parameter
+	 * used while dumping is considered. If the checksum config is to checksum the entire
+	 * file, then no changes need to be made since the checksum parameter isn't used anyway.
+	 */
+	if (opts.file_validation_chksm_config != FILE_VALIDATION_CHKSM_EVERY && 
```

Comment:
Space in the end of line.

If you use "git show" they would be highlighted red. (at least in my git setup)

Reply:
I didn't know that, thanks!